ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -62,6 +64,146 @@ thread_local bool ThreadAOTModInitialized = false;
 // Flag used to ensure that LLVM is initialised only once.
 once_flag LLVMInitialised;
 
+#ifndef NDEBUG
+// Left trim (in-place) the character `C` from the string `S`.
+void lTrim(string &S, const char C) {
+  S.erase(0, std::min(S.find_first_not_of(C), S.size() - 1));
+}
+
+// Dumps an LLVM Value to a string and trims leading whitespace.
+void dumpValueToString(Value *V, string &S) {
+  raw_string_ostream RSO(S);
+  V->print(RSO);
+  lTrim(S, ' ');
+}
+#endif
+
+enum DebugIR {
+  AOT,
```

Comment:
Is it guaranteed that these are given the numbers 0,1,2,3?

Reply:
I believe so:

>  If the first enumerator does not have an initializer, the associated value is zero. For any other enumerator whose definition does not have an initializer, the associated value is the value of the previous enumerator plus one. 

https://en.cppreference.com/w/cpp/language/enum

Reply:
Perfect!